great_expectations/rule_based_profiler/config/base.py

```diff
@@ -0,0 +1,214 @@
+import dataclasses
+from dataclasses import dataclass
+from typing import Any, Dict, List, Optional, Type
+
+from ruamel.yaml.comments import CommentedMap
+
+from great_expectations.data_context.types.base import BaseYamlConfig
+from great_expectations.marshmallow__shade import INCLUDE, Schema, fields, post_load
+from great_expectations.marshmallow__shade.decorators import post_dump
+from great_expectations.types import DictDot
+from great_expectations.util import filter_properties_dict
+
+
+class NotNullSchema(Schema):
+    """
+    Extension of Marshmallow Schema to facilitate implicit removal of null values before serialization.
+
+    The __config__ attribute is utilized to point a Schema to a configuration. It is the responsibility
+    of the child class to define its own __config__ to ensure proper serialization/deserialization.
+
+    Reference: https://marshmallow.readthedocs.io/en/stable/extending.html
+
+    """
+
+    @post_load
+    def make_config(self, data: dict, **kwargs) -> Type[DictDot]:
+        """Hook to convert the schema object into its respective config type.
+
+        Checks against config dataclass signature to ensure that unidentified kwargs are omitted
+        from the result object. This design allows us to maintain forwards comptability without
+        altering expected behavior.
+
+        Args:
+            data: The dictionary representation of the configuration object
+            kwargs: Marshmallow-specific kwargs required to maintain hook signature (unused herein)
+
+        Returns:
+            An instance of configuration class, which subclasses the DictDot serialization class
+
+        Raises:
+            NotImplementedError: If the subclass inheriting NotNullSchema fails to define a __config__
+
+        """
+        if not hasattr(self, "__config__"):
+            raise NotImplementedError(
+                "The subclass extending NotNullSchema must define its own custom __config__"
+            )
+
+        # Removing **kwargs before creating config object
+        recognized_attrs = {f.name for f in dataclasses.fields(self.__config__)}
+        cleaned_data = filter_properties_dict(
+            properties=data,
+            keep_fields=recognized_attrs,
+            clean_nulls=False,
+            clean_falsy=False,
+        )
+
+        return self.__config__(**cleaned_data)
+
+    @post_dump
+    def remove_nulls(self, data: dict, **kwargs) -> dict:
+        """Hook to clear the config object of any null values before being written as a dictionary.
+
+        Args:
+            data: The dictionary representation of the configuration object
+            kwargs: Marshmallow-specific kwargs required to maintain hook signature (unused herein)
+
+        Returns:
+            A cleaned dictionary that has no null values
+
+        """
+        cleaned_data = filter_properties_dict(
```

Comment:
@cdkini You might wish to consider the "deep" version -- just in case some config comes in as nested dictionaries, lists of ... etc. :)

Reply:
We only care about the surface level though right? We should be removing top level attrs before serialization but should not be modifying nested values.

```
self.batch_request = None # This should be removed from the object

# This should not be removed because it is indeed a non-null object (with null elements nested within)
self.batch_request = {
    "foo": None
}

```

Reply:
@cdkini I think you are right.  Thanks.